src/safariservices.cs

```diff
@@ -81,4 +118,141 @@ partial interface SFSafariViewControllerDelegate {
 		[Export ("safariViewController:didCompleteInitialLoad:")]
 		void DidCompleteInitialLoad (SFSafariViewController controller, bool didLoadSuccessfully);
 	}
+#else
+	[Mac (10,12)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface SFSafariApplication
+	{
+		[Static]
+		[Export ("getActiveWindowWithCompletionHandler:")]
+		void GetActiveWindow (Action<SFSafariWindow> completionHandler);
+
+		[Static]
+		[Export ("openWindowWithURL:completionHandler:")]
+		void OpenWindow (NSUrl url, [NullAllowed] Action<SFSafariWindow> completionHandler);
+
+		[Static]
+		[Export ("setToolbarItemsNeedUpdate")]
+		void SetToolbarItemsNeedUpdate ();
+	}
+
+	[Mac (10,12)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface SFSafariPage
+	{
+		[Export ("dispatchMessageToScriptWithName:userInfo:")]
+		void DispatchMessageToScript (string messageName, [NullAllowed] NSDictionary userInfo);
+
+		[Export ("reload")]
+		void Reload ();
+
+		[Export ("getPagePropertiesWithCompletionHandler:")]
+		void GetPageProperties (Action<SFSafariPageProperties> completionHandler);
+	}
+
+	[Mac (10,12)]
+	[Protocol, Model]
```

Comment:
I'm not sure about how this is used... but is a `[Model]` needed ? or is this something that will likely be adopted by an `NSView` (or similar) ?